xhr.bs

```diff
@@ -851,7 +836,8 @@ return <a>this</a>'s <a>cross-origin credentials</a>.
     </ol>
     <!-- upload complete flag can never be set here I hope -->
 
-    <p>To <a>process response</a> for <var>response</var>, run these steps:
+   <li>
+    <p>Let <var>processResponse</var>, given a <var>response</var>, be these steps:
```

Comment:
Some steps use "terminate these steps" to bail, others use "return". Probably "abort these steps" is most common and best?

Reply:
So I would kinda prefer if we could use return everywhere as it's a lot more convenient. And some of these algorithms can be invoked from in parallel and main thread alike.

Reply:
It's not a matter of parallel or main thread. It's a matter of which algorithm they terminate. "Return" as we use it today terminates the main algorithm...

Maybe it's unambiguous to allow it to terminate the innermost algorithm, but it seems a bit tricky.

Reply:
The thing I don't understand is why a step in a callback or a step in in parallel steps could ever terminate the main algorithm. It seems you would always have to handle that with some kind of flag.

Reply:
Well, some callbacks are called synchronously (e.g. readable stream read request steps), and so could theoretically return. But I agree it's an unlikely reading.

It's more that, when I go to scan an algorithm, I want to look for all "return"/"throw" lines and say "that's where the algorithm could terminate". If we allow re-using return to break out of a collection of steps, then I have to mentally discard those in my scan.

I don't have a strong argument for why this is more problematic for me in specs than it is in programming languages. Maybe the lack of syntax highlighting, or the fact that substeps for callbacks look very similar to substeps for if statements. But I hope this help explains my mild preference.